lai_s3/s3.py

```diff
@@ -1,41 +1,48 @@
 import botocore.exceptions
 import lightning as L
-from lightning.storage.payload import Payload
+import lightning_app as lapp
 import boto3
 import logging
 from typing import final, Union, Optional
+from torch.utils.data import Dataset, DataLoader
+from PIL import Image
+import io
+import json
 
 
 class S3(L.LightningWork):
 
     def __init__(
             self,
-            aws_access_key_id,
-            aws_secret_access_key,
+            aws_access_key_id=None,
+            aws_secret_access_key=None,
+            aws_session_token=None,
             *args, **kwargs
     ):
         super().__init__(self, *args, **kwargs)
 
         self.data = {} # Bucket name / bucket contents
-        self.aws_access_key_id = aws_access_key_id
-        self.aws_secret_access_key = aws_secret_access_key
+        self.credentials = {
+            'aws_access_key_id': aws_access_key_id,
+            'aws_secret_access_key': aws_secret_access_key,
+            'aws_session_token': aws_session_token
+        }
         self.verify_credentials()
 
     def verify_credentials(self):
+        missing_creds = []
+        for cred in self.credentials:
+            if self.credentials[cred] is None:
+                missing_creds += self.credentials[cred]
 
-        if sum([
-                self.aws_access_key_id is None,
-                self.aws_secret_access_key is None
-        ]) == 1:
-            missing_key = 'aws_access_key_id' if self.aws_secret_access_key \
-                else 'aws_secret_access_key'
+        if missing_creds:
             raise PermissionError(
-                "If either the aws_access_key_id or aws_secret_access_key is"
-                " provided then both are required."
-                f" Missing value for {missing_key}"
+                "If either the aws_access_key_id, aws_secret_access_key, or aws_session_token is"
```

Comment:
This shouldn't be true.  You should either need  `aws_access_key_id` and `aws_secret_access_key` or `aws_session_token` but not all three.

Reply:
actually this was actually the case when I tested. However, I am pushing a version that only uses the access key and id or the session token. Feel free to test perhaps I did something wrong.

Reply:
Further, this is also evident in the docs. Where did you see that session token was sufficient by itself my test + view of docs don't seem to agree with that.

The following example shows how you might set the environment variables for temporary security credentials and then call an AWS CLI command. Because no profile parameter is included in the AWS CLI command, the AWS CLI looks for credentials first in environment variables and therefore uses the temporary credentials.

Linux

```
$ export AWS_ACCESS_KEY_ID=ASIAIOSFODNN7EXAMPLE
$ export AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
$ export AWS_SESSION_TOKEN=AQoDYXdzEJr...<remainder of session token>
$ aws ec2 describe-instances --region us-west-1
```